lasy/profiles/longitudinal/longitudinal_profile_from_data.py

```diff
@@ -0,0 +1,151 @@
+    """
+    Derived class for longitudinal laser profile created using
+    data from an experimental measurement or from the output
+    of another code.
+
+    Uses user supplied data to define the longitudinal profile
+    of the laser pulse.
+
+    The data may be supplied in either the spectral or temporal
+    domain. The data should be passed as a structure (defined
+    below). If spectral data is passed, it will be converted to
+    the temporal domain.
+
+    Parameters
+    ----------
+    data : structure
+        The data structure comprises several items indexed by
+        a series of keys
+
+        datatype : string
+            The domain in which the data has been passed. Options
+            are 'spectral' and 'temporal'
+
+        abscissa : ndarrays of floats
+            The horizontal axis of the pulse duration measurement
+            When datatype is 'spectral' abscissa is wavelength in
+            meters
+            When datatype is 'temporal' abscissa is time in seconds
+
+        intensity : ndarrays of floats
+            The vertical axis of the pulse duration measurement
+            When datatype is 'spectral' intensity is spectral
+            intensity. This will be noramlized
+            When datatype is 'temporal' intensity is temporal
+            intensity. This will be normalized
+
+        phase : ndarrays of floats
+            If provided, this phase will be added to the pulse.
+            When datatype is 'spectral' phase is spectral
+            phase.
+            When datatype is 'temporal' phase is temporal
+            phase
+
+        dt : float
+            Only required when datatype is 'spectral'. In this
+            case this defines the user requested resolution in
+            the conversrion from the spectral to the temporal
+            domain.
+
+        wavelength : float
+            Only required when datatype is 'temporal'. Then,
+            this is the central wavelength of the pulse
+
+    lo, hi : floats (seconds)
+        Lower and higher end of the required domain of the data.
+        The data imported will be cut to this range prior to
+        being incorporated into the ``lasy`` pulse.
+    """
+
+    def __init__(self, data, lo, hi):
+        if data["datatype"] == "spectral":
+            # First find central frequency
+            wavelength = data["abscissa"]
+            spectral_intensity = data["intensity"]
+            spectral_phase = data["phase"]
+            dt = data["dt"]
+            cwl = np.sum(spectral_intensity * wavelength) / np.sum(spectral_intensity)
+            cfreq = c / cwl
+            # Determine required sampling frequency for desired dt
+            sample_freq = 1 / dt
+            # Determine number of points in temporal domain
+            indx = np.argmin(np.abs(wavelength - cwl))
+            dfreq = np.abs(c / wavelength[indx] - c / wavelength[indx + 1])
+            N = int((sample_freq) / dfreq)
```

Comment:
Also, the user could just specify the number of points in temporal domain, as an option. Would that be easy?
```suggestion
            N = int(sample_freq / dfreq)
```